uvq_pytorch/utils/distortionnet.py

```diff
@@ -0,0 +1,356 @@
+            self.load_state_dict(model_path)
+        if eval_mode:
+            self.model.eval()
+        self.num_patches_x = int(video_width / patch_width)
+        self.num_patches_y = int(video_height / patch_height)
+        self.feature_channels = feature_channels
+        self.feature_height = feature_height
+        self.feature_width = feature_width
+        self.patch_width = patch_width
+        self.patch_height = patch_height
+        self.video_width = video_width
+        self.video_height = video_height
+        self.video_channels = video_channels
+        self.depth = depth
+        self.patch_feature_height = int(feature_height / self.num_patches_y)
+        self.patch_feature_width = int(feature_width / self.num_patches_x)
+        self.label_dim = label_dim
+
+    def load_state_dict(self, model_path) -> nn.Module:
+        model = torch.load(model_path)
+        self.model.load_state_dict(model)
+        return model
+
+    def predict(self, frame):
+        with torch.no_grad():
+            _, label_probs = self.model(Tensor(frame))
+        return label_probs.detach().numpy()
+
+    def predict_and_get_features(self, frame) -> tuple[np.ndarray, np.ndarray]:
+        with torch.no_grad():
+            features, label_probs = self.model(torch.Tensor(frame))
+        return (
+            features.detach().numpy(),
+            label_probs.detach().numpy(),
+        )
+
+    def get_labels_and_features_for_all_frames(
+        self,
+        video,
+    ):
+        # TODO: allow converting video to a batch of patches and running batch prediction instead of sliding window for loops
+        label = np.ndarray(
+            (video.shape[0], self.num_patches_y, self.num_patches_x, self.label_dim),
+            np.float32,
+        )
+        feature = np.ndarray(
+            (
+                video.shape[0],
+                self.feature_height,
+                self.feature_width,
+                self.feature_channels,
+            ),
+            np.float32,
+        )
+
+        if self.depth == 1:
+            # TODO: this is for distortionnet; for compressionnet keep only the "else" part.
+            patch = np.ndarray(
+                (1, self.video_channels, self.patch_height, self.patch_width),
+                np.float32,
+            )
+        else:
+            video = video.transpose(0, 2, 1, 3, 4)
+            patch = np.ndarray(
+                (
+                    1,
+                    self.video_channels,
+                    self.depth,
+                    self.patch_height,
+                    self.patch_width,
+                ),
+                np.float32,
+            )
+
+        for k in range(video.shape[0]):
+            for j in range(self.num_patches_y):
+                for i in range(self.num_patches_x):
+                    if self.depth == 1:
+                        # TODO: this would work for distortionnet, it's not used here and should be cleaned up; keep only the "else" part
```

Comment:
Same here. Remove this comment and remove the "else" part (the "if" is also redundant if there is only 1 branch).